sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -515,20 +524,105 @@ export class ApplicationOfferingChangeRequestService {
    * @param studentConsent student consent to approve the application offering change request.
+   * @param userId user id of the user making the request.
    */
   async updateApplicationOfferingChangeRequest(
     applicationOfferingChangeRequestId: number,
     applicationOfferingChangeRequestStatus: ApplicationOfferingChangeRequestStatus,
     studentConsent: boolean,
+    userId: number,
   ): Promise<void> {
+    const auditUser = { id: userId } as User;
+    const currentDate = new Date();
     await this.applicationOfferingChangeRequestRepo.update(
       {
         id: applicationOfferingChangeRequestId,
       },
       {
         applicationOfferingChangeRequestStatus,
         studentConsent,
+        studentActionDate: currentDate,
+        modifier: auditUser,
+        updatedAt: currentDate,
       },
     );
   }
+
+  /**
+   * Update the application offering change request status for the given application offering change request id.
+   * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
+   * @param applicationId application id related to the application offering change request.
+   * @param offeringId requested offering id related to this application offering change request.
+   * @param studentId student id of the student related to the application offering change request.
+   * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
+   * @param assessmentNote note added while updating the application offering change request.
+   * @param userId user id of the user making the approve or decline request.
+   */
+  async assessApplicationOfferingChangeRequest(
+    applicationOfferingChangeRequestId: number,
+    applicationId: number,
+    offeringId: number,
+    studentId: number,
+    applicationOfferingChangeRequestStatus: ApplicationOfferingChangeRequestStatus,
+    assessmentNote: string,
+    userId: number,
+  ): Promise<void> {
+    const auditUser = { id: userId } as User;
+    const currentDate = new Date();
+    const application = new Application();
+    application.id = applicationId;
+    // Create a new assessment if the application offering change request status is approved.
+    if (
+      applicationOfferingChangeRequestStatus ===
+      ApplicationOfferingChangeRequestStatus.Approved
+    ) {
+      application.currentAssessment = {
+        application: { id: applicationId } as Application,
+        triggerType: AssessmentTriggerType.ApplicationOfferingChange,
+        offering: { id: offeringId } as EducationProgramOffering,
+        applicationOfferingChangeRequest: {
+          id: applicationOfferingChangeRequestId,
+        } as ApplicationOfferingChangeRequest,
+        creator: auditUser,
+        createdAt: currentDate,
+        submittedBy: auditUser,
+        submittedDate: currentDate,
+      } as StudentAssessment;
+    }
```

Comment:
shouldn't this also be part of the transaction?

Reply:
In IMO, this part should happen after the application offering change update in the same transaction, unless there is a dependency